### PR TITLE
Fix Health Checks Dependencies

### PR DESCRIPTION
**Pull request by rajvi**

### CODE REVIEW

**Review by rafsalas19 — CHANGES_REQUESTED** [review]
So, the install packages are named differently between ubuntu 20.04 and 22.04.
Let's just remove this dependency all together. i.e remove all instances of mlocate and plocate from the install script. Then on this file: azurehpc-health-checks/customTests/azure_common.nhc replace "hpcx=$(locate hpcx-init.sh)" with ```hpcx=$(find /opt/hpcx* -maxdepth 1 -name "hpcx-init.sh")```

File: install-nhc.sh [hunk 1]
```diff
@@ -82,12 +82,16 @@ else
...
+# Install locate
+sudo apt-get install mlocate
 # Check cmake version + install if necessary
 output=$(cmake --version | sed -n 1p | sed 's/[^0-9]*//g')
 export NHC_CMAKE=cmake
 if [ $output -lt 3200 ]; then
     echo "Upgrade cmake version to 3.20 or above to build nvbandwidth"
     pushd $SRC_DIR/build
+      # Install Boost for CMake
+      sudo apt install build-essential libboost-system-dev libboost-thread-dev libboost-program-options-dev libboost-test-dev
```
Comment (rafsalas19) [inline install-nhc.sh]: This is going to break the Alma installation. Line 73 is where the installation should be happening because Ubuntu is a Debian based Linux distribution. Also, whenever we use apt-get or apt install in a script you have to use the confirm argument "-y" or else the apt installer expects you to manually confirm.

**Review by rafsalas19 — CHANGES_REQUESTED** [review]

File: install-nhc.sh (hunk 1 above)
Comment (rajvi) [reply install-nhc.sh]: Ok, I'll add installing Boost only for ubuntu 20.04 which is the image that had this error. I added "yes" on the script that calls install-nhc.sh. I'll do this instead.

File: install-nhc.sh [hunk 2]
```diff
@@ -85,9 +85,19 @@ fi
 # Check cmake version + install if necessary
 output=$(cmake --version | sed -n 1p | sed 's/[^0-9]*//g')
 export NHC_CMAKE=cmake
+# Find VM OS
```
Comment (rafsalas19) [inline install-nhc.sh]: Lets not do lines 88-93 and lines 97-100 . You can determine the OS by looking at this file: /etc/os-release. 

Please revert this change.

File: install-nhc.sh [hunk 3]
```diff
@@ -70,12 +70,12 @@ distro_check=$( cat /etc/os-release | grep -i ID_LIKE=)
 distro=`awk -F= '/^NAME/{print $2}' /etc/os-release`
 
 if [[ $distro_check =~ "debian" ]]; then
-  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y libpci-dev hwloc build-essential libboost-program-options-dev libssl-dev plocate cmake
+  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y libpci-dev hwloc build-essential libboost-program-options-dev libssl-dev cmake $(apt-cache search libpci-dev | grep -o '^[^ ]*')
 elif [[ $distro =~ "AlmaLinux" ]]; then
-  sudo dnf install -y pciutils-devel hwloc openssl-devel boost-devel mlocate cmake
+  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y pciutils-dev hwloc libssl-dev libboost-dev cmake
```
Comment (rafsalas19) [inline install-nhc.sh]: Apt-get does not work on AlmaLinux. Please revert to the old way of using dnf.

add whatever additional packages to the existing install line (previously line 75)

Please test this change on AlmaLinux

Comment (rafsalas19) [reply install-nhc.sh]: Obviously remove any references to mlocate/plocate

File: install-nhc.sh [hunk 4]
```diff
@@ -70,12 +70,12 @@ distro_check=$( cat /etc/os-release | grep -i ID_LIKE=)
 distro=`awk -F= '/^NAME/{print $2}' /etc/os-release`
 
 if [[ $distro_check =~ "debian" ]]; then
```
Comment (rafsalas19) [inline install-nhc.sh:72]: This condition targets both Ubuntu 20.0 and 22.04 

I believe the boost install should be the same for both 20.04/22.04 

So, we most likely don't need to differentiate between the two.

Please try to add any additional boost installs to the previous line 73

File: install-nhc.sh [hunk 5]
```diff
@@ -70,12 +70,12 @@ distro_check=$( cat /etc/os-release | grep -i ID_LIKE=)
...
-  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y libpci-dev hwloc build-essential libboost-program-options-dev libssl-dev plocate cmake
+  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y libpci-dev hwloc build-essential libboost-program-options-dev libssl-dev cmake $(apt-cache search libpci-dev | grep -o '^[^ ]*')
 elif [[ $distro =~ "AlmaLinux" ]]; then
-  sudo dnf install -y pciutils-devel hwloc openssl-devel boost-devel mlocate cmake
+  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y pciutils-dev hwloc libssl-dev libboost-dev cmake
 # this will attempt to install for the following os's: CentOS, RHEL, Fedora
 elif [[ $distro_check =~ "CentOS" ]] || [[ $distro_check =~ "Rhel" ]] || [[ $distro_check =~ "fedora" ]]; then
-  sudo yum install -y  pciutils-devel hwloc openssl-devel boost-devel mlocate cmake > /dev/null
+  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y pciutils-dev hwloc libssl-dev libboost-dev cmake > /dev/null
```
Comment (rafsalas19) [inline install-nhc.sh]: apt-get does not work on these distribution types. Please revert the change and use the previous line excluding the locate package. Yum is the package manager we want to use.

File: install-nhc.sh (hunk 4 above)
Comment (rafsalas19) [reply install-nhc.sh:72]: What does this do ```$(apt-cache search libpci-dev | grep -o '^[^ ]*')```?

**Review by rafsalas19 — CHANGES_REQUESTED** [review]

File: install-nhc.sh (hunk 2 above)
Comment (rajvi) [reply install-nhc.sh]: Do we want to determine the OS using the other option in order to install Boost? Or remove finding the OS altogether?

File: install-nhc.sh (hunk 4 above)
Comment (rajvi) [reply install-nhc.sh:72]: If installing boost will break Alma, do we want to install boost only on Ubuntu 20/22?

Comment (rajvi) [reply install-nhc.sh:72]: That searches the package cache for containing the string "libpci-dev"

Comment (rajvi) [reply install-nhc.sh:72]: Removed locate as discussed

File: install-nhc.sh (hunk 3 above)
Comment (rajvi) [reply install-nhc.sh]: Removed locate as discussed

**Review by rafsalas19 — APPROVED** [review]
